src/amms.py

```diff
@@ -7,297 +7,609 @@
-			LOG.error("Request to token API timed out")
-			self.get_token_list()
-
-	def store_pools(self):
-		"""
-		Save pools data to database.
-		"""
-		self.get_token_list()
-		for pool in self.pools:
-			self.store_pool(pool)
-
-	def store_pool(self, pool: Dict[str, Any]) -> None:
-		"""
-		Save pool data to database.
-		"""
-		market_address = pool.get('id', 'Unknown')
-
-		# get token symbols
-		try:
-			token_x_symbol = next(i for i in self.token_list if i['mint'] == pool['mintA']).get('symbol')
-		except StopIteration:
-			token_x_symbol = 'Unknown'
-		try:
-			token_y_symbol = next(i for i in self.token_list if i['mint'] == pool['mintB']).get('symbol')
-		except StopIteration:
-			token_y_symbol = 'Unknown'
-
-		pair = f"{token_x_symbol}-{token_y_symbol}"
-
-		# Create a new AmmLiquidity record
-		with get_db_session() as session:
-			liquidity_entry = AmmLiquidity(
-				timestamp=self.timestamp,
-				dex=self.DEX_NAME,
-				pair=pair,
-				market_address=market_address,
-				additional_info=json.dumps(pool)
-			)
-
-			# Add the new record to the session and commit
-			session.add(liquidity_entry)
-			session.commit()
+    DEX_NAME = "Orca"
+    token_list: List
+
+    def get_pools(self):
+        """
+        Fetches pool data from the Radium API and stores it in the `pools` attribute.
+        """
+        LOG.info("Fetching pools from Raydium API")
+        try:
+            result = requests.get(
+                "https://api.raydium.io/v2/ammV3/ammPools", timeout=30
+            )
+            decoded_result = result.content.decode("utf-8")
+            self.pools = json.loads(decoded_result)["data"]
+            LOG.info(f"Successfully fetched {len(self.pools)} pools")
+        except requests.exceptions.Timeout:
+            LOG.error("Request to Raydium pool API timed out")
+            self.get_pools()
+
+    def get_token_list(self) -> None:
+        """
+        Get list of token with symbols from Orca API.
+        """
+        LOG.info("Fetching token info API")
+        try:
+            result = requests.get(
+                "https://api.mainnet.orca.so/v1/token/list", timeout=30
+            )
+            decoded_result = result.content.decode("utf-8")
+            self.token_list = json.loads(decoded_result)["tokens"]
+        except requests.exceptions.Timeout:
+            LOG.error("Request to token API timed out")
+            self.get_token_list()
+
+    def store_pools(self):
+        """
+        Save pools data to database.
+        """
+        self.get_token_list()
+        for pool in self.pools:
+            self.store_pool(pool)
+
+    def store_pool(self, pool: Dict[str, Any]) -> None:
+        """
+        Save pool data to database.
+        """
+        market_address = pool.get("id", "Unknown")
+
+        # get token symbols
+        try:
+            token_x_symbol = next(
+                i for i in self.token_list if i["mint"] == pool["mintA"]
+            ).get("symbol")
```

Comment:
@nickgore not really sure what is going on there

Reply:
Understood. Since I'm doing the rehaul of how we process data on Starknet, let's discuss whether we should even have symbols or addresses in the db tomorrow :slightly_smiling_face: